types/TestResult.js

```diff
@@ -100,20 +100,20 @@ export type FormattedAssertionResult = {
 
 export type AggregatedResult = {
   coverageMap?: ?CoverageMap,
-  numFailedTests: number,
-  numFailedTestSuites: number,
-  numPassedTests: number,
-  numPassedTestSuites: number,
-  numPendingTests: number,
-  numPendingTestSuites: number,
-  numRuntimeErrorTestSuites: number,
-  numTotalTests: number,
-  numTotalTestSuites: number,
-  snapshot: SnapshotSummary,
-  startTime: number,
+  numFailedTests?: number,
+  numFailedTestSuites?: number,
+  numPassedTests?: number,
+  numPassedTestSuites?: number,
+  numPendingTests?: number,
+  numPendingTestSuites?: number,
+  numRuntimeErrorTestSuites?: number,
+  numTotalTests?: number,
+  numTotalTestSuites?: number,
+  snapshot?: SnapshotSummary,
+  startTime?: number,
   success: boolean,
-  testResults: Array<TestResult>,
-  wasInterrupted: boolean,
+  testResults?: Array<TestResult>,
+  wasInterrupted?: boolean,
```

Comment:
Can you elaborate why you are making all fields optional? It leads to more complex code. I don't think most of these are nullable. The only place we create the results objects is here: https://github.com/facebook/jest/blob/c6af39fd7f63f30f97c8974d266deb185bf7b116/packages/jest-cli/src/TestRunner.js#L389 and it clearly initializes the values.